### PR TITLE
remove unnecessary reflection, fix crafting request not working in wireless terminal

### PR DESCRIPTION
**Pull request by Alexdoru**

### CODE REVIEW

**Comment by Glease** [comment]
@vfyjxf hi, does this look good? can't put you on reviewers list somehow.

**Comment by Alexdoru** [comment]
I tested in full pack and it's working, no crash and the request crafting missing items is working again from the wireless terminal

**Comment by Dream-Master** [comment]
i pm vfyjxf he not respond

**Comment by vfyjxf** [comment]
> i pm vfyjxf he not respond

Sorry, I've been mostly inside school in recent months and can only return on weekends to check for information.

File: src/main/java/com/github/vfyjxf/nee/NEINeeConfig.java [hunk 1]
```diff
@@ -139,11 +138,6 @@ private void installThaumicEnergisticsSupport() {
     }
 
     private void installPatternTerminalExSupport(Set<String> identifiers) {
-        try {
```
Comment (eigenraven) [inline src/main/java/com/github/vfyjxf/nee/NEINeeConfig.java:142]: Why did you remove this check? You're converting a soft dependency to a hard dependency by doing this

File: src/main/java/com/github/vfyjxf/nee/utils/GuiUtils.java [hunk 2]
```diff
@@ -16,106 +20,57 @@
...
-
-        try {
-            Class<?> guiWirelessCraftingClass =
-                    Class.forName("net.p455w0rd.wirelesscraftingterminal.client.gui.GuiWirelessCraftingTerminal");
-            return guiWirelessCraftingClass.isInstance(gui);
-        } catch (ClassNotFoundException e) {
-            return false;
-        }
+        if (!isWirelessCraftingTerminalModLoaded) return false;
+        return gui instanceof net.p455w0rd.wirelesscraftingterminal.client.gui.GuiWirelessCraftingTerminal;
```
Comment (eigenraven) [inline src/main/java/com/github/vfyjxf/nee/utils/GuiUtils.java:27]: This will make class loading fail if WCT is not present, have you tested this mod to work without it?

File: src/main/java/com/github/vfyjxf/nee/NEINeeConfig.java (hunk 1 above)
Comment (Alexdoru) [reply src/main/java/com/github/vfyjxf/nee/NEINeeConfig.java:142]: image

applied energistics is already a hard dependency

File: src/main/java/com/github/vfyjxf/nee/utils/GuiUtils.java (hunk 2 above)
Comment (Alexdoru) [reply src/main/java/com/github/vfyjxf/nee/utils/GuiUtils.java:27]: if the mod isn't loaded it returns false and doesn't reach the part that would throw a class not found exception, I just tested it and it works
image

File: src/main/java/com/github/vfyjxf/nee/NEINeeConfig.java (hunk 1 above)
Comment (eigenraven) [reply src/main/java/com/github/vfyjxf/nee/NEINeeConfig.java:142]: terminalex doesn't exist in vanilla ae2 so this is for compat with non-gtnh ae2 which I guess we don't really need to keep?

**Review by eigenraven — APPROVED** [review]

File: src/main/java/com/github/vfyjxf/nee/NEINeeConfig.java (hunk 1 above)
Comment (Alexdoru) [reply src/main/java/com/github/vfyjxf/nee/NEINeeConfig.java:142]: ah didn't know about that but yeah let it crash >:D
